handler.go

```diff
@@ -258,16 +258,6 @@ func (h *Handler) ServeHTTP(responseWriter http.ResponseWriter, request *http.Re
 	if timeoutErr != nil {
 		clientVisibleError = timeoutErr
 	}
-	// If NewStream or SetTimeout errored and the protocol doesn't want the
-	// error sent to the client, sender and/or receiver may be nil. We still
-	// want the error to be seen by interceptors, so we provide no-op Sender
-	// and Receiver implementations.
-	if clientVisibleError != nil && sender == nil {
-		sender = newNopSender(h.spec, responseWriter.Header(), make(http.Header))
-	}
-	if clientVisibleError != nil && receiver == nil {
-		receiver = newNopReceiver(h.spec, request.Header, request.Trailer)
-	}
```

Comment:
These were showing up as uncovered by tests - looking at the implementation the sender/receiver are never nil:

* Connect: https://github.com/bufbuild/connect-go/blob/c9b9bcbabc808ef86e5bedf0d080916c3047df8d/protocol_connect.go#L203-L212
* gRPC: https://github.com/bufbuild/connect-go/blob/c9b9bcbabc808ef86e5bedf0d080916c3047df8d/protocol_grpc.go#L173-L180